pkgs/development/python-modules/beanhub-cli/default.nix

```
@@ -55,12 +71,35 @@ buildPythonPackage rec {
     rich
     starlette-wtf
     uvicorn
-  ];
+  ] ++ optional-dependencies.all;
```

```suggestion
  ];
```

Why? optional-dependencies are optional.

---

Because beanhub-cli is a command line tool project instead of a library intend for others to import and use. As far as I know, installing optional dependencies with pip is easy as like this

```bash
pip install "beanhub-cli[connect]"
```

But with nix, it's much harder. For example, I can run beanhub-cli with nix easily

```bash
nix-shell -p beanhub-cli
```

I wish there's a way to install the connect dependencies

```bash
nix-shell -p "beanhub-cli[connect]"
```

However, obviously it doesn't work that way for nix.

It's mostly about the user experience. And since most people use beanhub-cli requires the `connect` feature, so we are thinking about making it just the default dependency anyway in the future release.

Hope that makes sense.